libbeat/processors/script/javascript/session.go

```diff
@@ -288,14 +289,28 @@ func newSessionPool(p *goja.Program, c Config) (*sessionPool, error) {
 			s, _ := newSession(p, c, false)
 			return s
 		},
-		C: make(chan *session, c.MaxCachedSessions),
+		C:                  make(chan *session, c.MaxCachedSessions),
+		NewSessionsAllowed: !c.OnlyCachedSessions,
 	}
 	pool.Put(s)
 
+	// If we are not allowed to create new sessions, pre-cache requested sessions
+	if !pool.NewSessionsAllowed {
+		for i := 0; i < c.MaxCachedSessions-1; i++ {
+			pool.Put(pool.New())
+		}
+	}
+
 	return &pool, nil
 }
 
 func (p *sessionPool) Get() *session {
+
+	if !p.NewSessionsAllowed {
+		return <-p.C
+	}
```

Comment:
can this be moved to the select statement?  I haven't tried it, but it seems like it could be.

Reply:
yeah that gets a `InvalidSelectCase occurs when a select case is not a channel send or receive.` I can move it into the default statement but based on our quick slack chat ill leave it like this